actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -609,7 +609,7 @@ def mount(app, options = nil)
           target_as       = name_for_action(options[:as], path)
           options[:via] ||= :all
 
-          match(path, options.merge(to: app, anchor: false, format: false))
+          match(path, options.reverse_merge(to: app, anchor: false, format: false))
```

Comment:
This is changing the behavior of mount, so I think we need a test for it. Also, can we just change the order of the hash instead of using `reverse_merge`?

Reply:
> Also, can we just change the order of the hash instead of using `reverse_merge`?

You mean like this?

`{ to: app, anchor: false, format: false }.merge(options)`

I think that does the same thing as `reverse_merge`, so I'm not sure I understand your suggestion.

Reply:
Added a test 👍

Reply:
Yes, exactly that one. `reverse_merge` is confusing in my opinion, and also requires the framework to load that core-ext, that I don't think we are doing right now, at least I can't find actionpack requiring that core ext.

Reply:
Cool, I changed that around.